src/components/types.ts

```diff
@@ -238,3 +238,14 @@ export type MultimodalInputProps = TextInputProps &
     | 'filePreviewsContainer'
     | 'errorMessagesContainer'
   >
+
+export interface QuestionFormat extends DataFormat {
+  options: (string | number)[]
+}
+
+export interface QuestionProps extends QuestionFormat {
+  ws: WebSocketClient
+  messageId: string
+  conversationId: string
+  currentUser: string
```

Comment:
I also think that we should extract these in different interface so that they can be used with other components as required

Reply:
I wanted to distinguish between sender and currentUser since I figured there may be times where the original sender info might be needed vs the currentUser. In this case, the currentUser is needed to reply back to the original message. What do you think?

Reply:
yes the sender that elementRenderer passes down must always be the current user. That is same for the other input elements. Refer `TextInput` and `MultimodalInput`